app/assets/javascripts/discourse/tests/integration/components/plugin-outlet-test.js

```diff
@@ -103,6 +113,71 @@ module("Integration | Component | plugin-outlet", function (hooks) {
     );
   });
 
+  test("Can act as a wrapper around core implementation", async function (assert) {
+    const consoleErrorStub = sinon.stub(console, "error");
+
+    this.set("shouldDisplay", false);
+    this.set("yieldCore", false);
+    this.set("enableClashingConnector", false);
+
+    extraConnectorClass("outlet-with-default/my-connector", {
+      shouldRender(args) {
+        return args.shouldDisplay;
+      },
+    });
+
+    extraConnectorClass("outlet-with-default/clashing-connector", {
+      shouldRender(args) {
+        return args.enableClashingConnector;
+      },
+    });
+
+    await render(
+      hbs`
+        <PluginOutlet @name="outlet-with-default" @outletArgs={{hash shouldDisplay=this.shouldDisplay yieldCore=this.yieldCore enableClashingConnector=this.enableClashingConnector}}>
+          <span class='result'>Core implementation</span>
+        </PluginOutlet>
+      `
+    );
+
+    assert.dom(".result").hasText("Core implementation");
+
+    this.set("shouldDisplay", true);
+    await settled();
+
+    assert.dom(".result").hasText("Plugin implementation");
+
+    this.set("yieldCore", true);
+    await settled();
+
+    assert.dom(".result").hasText("Plugin implementation Core implementation");
+
+    assert.strictEqual(consoleErrorStub.callCount, 0, "no errors in console");
+
+    this.set("enableClashingConnector", true);
+    await settled();
+
+    assert.strictEqual(
+      consoleErrorStub.callCount,
+      1,
+      "clash error reported to console"
+    );
```

Comment:
You could have another error preventing the other to happen and making you think it's working correctly. Maybe we should try to check the actual message?